pyiceberg/table/__init__.py

```diff
@@ -3537,6 +3537,58 @@ def update_partitions_map(
             schema=table_schema,
         )
 
+    def files(self) -> "pa.Table":
```

Comment:
Could we add the snapshot_id as an optional argument to support time travel?

Here's an example of how we are doing it for the [Partitions metadata table](https://github.com/apache/iceberg-python/blob/main/pyiceberg/table/__init__.py#L3452)